### PR TITLE
docs/editor: about the nvim setup

### PR DESCRIPTION
**Pull request by Ruixi-rebirth**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -21,7 +21,14 @@
         packages.nixd = nixd;
 
         devShells.default = nixd.overrideAttrs (old: {
-          nativeBuildInputs = old.nativeBuildInputs ++ [ pkgs.clang-tools pkgs.gdb pkgs.nixpkgs-fmt ];
+          nativeBuildInputs = old.nativeBuildInputs ++ [
+            pkgs.clang-tools
+            pkgs.gdb
+            pkgs.nixpkgs-fmt
+          ] ++ [
```
Comment (inclyc) [inline flake.nix:28]: how about separate the shell here in something like `devShells.nvim` ?

File: docs/editor-setup.md [hunk 2]
```diff
@@ -55,3 +55,10 @@ Please file a bug if you encountered some trouble using the extension.
 
 Neovim native LSP and [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig).
 We are officially supported by nvim-lspconfig, see [upstream docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.txt#nixd)
+
+also, the [example config](https://github.com/nix-community/nixd/blob/main/editors/nvim-lsp.nix) for testing, You can run the following command to edit a *.nix file
```
Comment (inclyc) [inline docs/editor-setup.md:59]:
```suggestion
Also, we provide an [example config](/editors/nvim-lsp.nix) for testing, You can run the following command to edit a *.nix file
```

File: docs/editor-setup.md [hunk 3]
```diff
@@ -55,3 +55,10 @@ Please file a bug if you encountered some trouble using the extension.
...
+```bash
```
Comment (inclyc) [inline docs/editor-setup.md:60]:
```suggestion
```console
```

File: flake.nix (hunk 1 above)
Comment (Ruixi-rebirth) [reply flake.nix:28]: Maybe it's better this way

File: docs/editor-setup.md [hunk 4]
```diff
@@ -55,3 +55,17 @@ Please file a bug if you encountered some trouble using the extension.
...
 Neovim native LSP and [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig).
 We are officially supported by nvim-lspconfig, see [upstream docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.txt#nixd)
+
+Also, we provide an [example config](/editors/nvim-lsp.nix) for testing, You can run the following command to edit a *.nix file
+```console
+$ nix develop github:nix-community/nixd#nvim
+
+$ nvim-lsp /tmp/test/default.nix
+```
+tip: If you want to configure lsp itself, see [configuration](/docs/user-guide.md#configuration), and the following tree-like directory
```
Comment (inclyc) [inline docs/editor-setup.md:65]: > and the following tree-like directory

is this grammar correct?

**Review by inclyc — APPROVED** [review]
nitpick for English grammar, others LGTM and works perfect on my machine

**Pull request merged**